webui/react/src/components/LogViewer/LogViewerFilters.tsx

```diff
@@ -24,48 +25,72 @@ export interface Filters {
   stdtypes?: string[],
 }
 
-const TaskLogFilters: React.FC<Props> = ({ onChange, onReset, options, values }: Props) => {
+const LogViewerFilters: React.FC<Props> = ({ onChange, onReset, options, values }: Props) => {
   const selectOptions = useMemo(() => {
+    const { agentIds, allocationIds, containerIds, rankIds } = options;
     return {
       ...options,
+      agentIds: agentIds ? agentIds.sortAll(alphaNumericSorter) : undefined,
+      allocationIds: allocationIds ? allocationIds.sortAll(alphaNumericSorter) : undefined,
+      containerIds: containerIds ? containerIds.sortAll(alphaNumericSorter) : undefined,
       levels: Object.entries(LogLevelFromApi)
         .filter(entry => entry[1] !== LogLevelFromApi.Unspecified)
         .map(([ key, value ]) => ({ label: key, value })),
+      rankIds: rankIds ? rankIds.sortAll(alphaNumericSorter) : undefined,
     };
   }, [ options ]);
 
+  const show = useMemo(() => {
+    return Object.keys(selectOptions).reduce((acc, key) => {
+      const filterKey = key as keyof Filters;
+      const options = selectOptions[filterKey];
+
+      // !! casts `undefined` into the boolean value of `false`.
+      acc[filterKey] = !!(options && options.length > 1);
+
+      return acc;
+    }, {} as Record<keyof Filters, boolean>);
+  }, [ selectOptions ]);
+
+  const isResetShown = useMemo(() => {
+    const keys = Object.keys(selectOptions);
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i] as keyof Filters;
+      const value = values[key];
+      if (value && value.length !== 0) return true;
+    }
+    return false;
+  }, [ selectOptions, values ]);
+
   const handleChange = useCallback((
     key: keyof Filters,
     caster: NumberConstructor | StringConstructor,
   ) => (value: SelectValue) => {
-    onChange?.({
-      ...values,
-      [key]: (value as Array<string>).map(item => caster(item)),
-    });
+    onChange?.({ ...values, [key]: (value as Array<string>).map(item => caster(item)) });
   }, [ onChange, values ]);
 
   const handleReset = useCallback(() => onReset?.(), [ onReset ]);
 
   return (
     <>
       <Space>
-        {selectOptions?.allocationIds?.length !== 0 && (
+        {show.allocationIds && (
```

Comment:
would it make sense to do:
<img width="456" alt="image" src="https://user-images.githubusercontent.com/28383080/156834147-2e30b842-6042-43e3-9bf8-861d7fc7b6c9.png">

instead for these? or maybe rename `show` to `moreThanOne` ?

Reply:
That way we can get rid of show. Or are there rendering considerations??

Reply:
👍 

re-rendering wise it's the same because they both depend on `selectOptions` changes.

I think your suggestion makes sense, and it'll reduce the line count by a little.

Reply:
actually thought about it a bit more and the current method centralizes the show logic into one spot, so I'll follow your second suggestion instead ;)